src/services/Git.ts

```diff
@@ -27,17 +44,88 @@ export class Git {
 			'--short',
 			'HEAD'
 		];
-		const branch: string = await executeTerminalCommand(GIT_COMMAND, args);
-		return branch && branch.trim();
+		const { stdout: branch } = await this.execFile({ args });
+		return branch && branch.toString().trim();
+	}
+
+	public async push(branch: string, forceUpstream?: boolean) {
```

Comment:
possibly pass `args` argument here instead of boolean that could include `--set-upstream` as an example.